src/blockdb/wrapper.h

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Copyright (c) 2015-2018 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "blockdb.h"
+#include "main.h"
+#include "undo.h"
+
+enum FlushStateMode
+{
+    FLUSH_STATE_NONE,
+    FLUSH_STATE_IF_NEEDED,
+    FLUSH_STATE_PERIODIC,
+    FLUSH_STATE_ALWAYS
+};
+
+enum BlockDBMode
+{
+    SEQUENTIAL_BLOCK_FILES, // 0
+    DB_BLOCK_STORAGE // 1
+};
+
+static const BlockDBMode DEFAULT_BLOCK_DB_MODE = SEQUENTIAL_BLOCK_FILES;
+extern BlockDBMode BLOCK_DB_MODE;
+
+/** Determine if the block db mode we started with is behind another one already on disk*/
+bool DetermineStorageSync();
+
+/** Catch leveldb up with sequential block files */
+void SyncStorage(const CChainParams &chainparams);
+
+/** Functions for disk access for blocks */
+bool ReadBlockFromDisk(CBlock &block, const CBlockIndex *pindex, const Consensus::Params &consensusParams);
+bool WriteBlockToDisk(const CBlock &block, CDiskBlockPos &pos, const CMessageHeader::MessageStartChars &messageStart);
+
+bool UndoWriteToDisk(const CBlockUndo &blockundo,
+    CDiskBlockPos &pos,
+    const CBlockIndex *pindex,
+    const CMessageHeader::MessageStartChars &messageStart);
+bool UndoReadFromDisk(CBlockUndo &blockundo, const CDiskBlockPos &pos, const CBlockIndex *pindex);
```

Comment:
These APIs have been nicely isolated and constitute the interface into block storage systems.  I would like to see them carefully inline documented, including descriptions of all parameters, and return values.  

Where it makes sense, also change the parameter names for clarity.  In particular, UndoWriteToDisk's "pindex" parameter is not a pointer to the CBlockIndex whose undo information is being written, but a pointer to the previous block.

Also, I think that this API writes/reads Undo information to the disk.  It does not undo a write to disk.  Let's call them WriteUndoToDisk/ReadUndoFromDisk.  This is consistent with WriteBlockToDisk/ReadBlockFromDisk. 

Yes, I see that they were poorly named prior to this PR, but let's fix now since they are not hidden in "main" but now APIs.